state/machine.go

```diff
@@ -1111,22 +1111,25 @@ func (m *Machine) DesiredSpaces() (set.Strings, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	// We ignore negative spaces as it doesn't change what spaces we do want.
```

Comment:
What if I only set a negative space? Surely it should set all the other spaces as positive options in no positive options were set?